Add VPN configuration details for different cloud providers (#3806)
astronomer/docs: astro/connect-azure.md

    groupId="create-a-private-connection-between-astro-and-azures"
    values={[
        {label: 'VNet peering', value: 'VNet peering'},

>         {label: 'VHub peering', value: 'VHub peering'},

Reviewer: add to list of options on line 33

        {label: 'Azure Private Link', value: 'Azure Private Link'},

>         {label: 'VPN', value: 'VPN'},

Reviewer: add to list of options on line 33

To set up a private connection between an Astro Virtual Network (VNet) and an Azure VNet, you can create a VNet peering connection. VNet peering ensures private and secure connectivity, reduces network transit costs, and simplifies network layouts.

1. Retrieve the following information from the target Azure environment that you want to connect with:

    - Azure Tenant ID and Subscription ID.
    - VNet ID.
    - Resource Group ID.

> 2. Prepare JSON file `astro-vnet-peering-creator-role.json` with the following permisisons (`{customer-subscription-id}` has to be replaced with your value):

Reviewer 2: ```suggestion
2. Prepare the `astro-vnet-peering-creator-role.json` JSON file with the following permissions. Replace `{customer-subscription-id}` with your value:
```

            "Microsoft.Network/virtualNetworks/virtualNetworkPeerings/write",
            "Microsoft.Network/virtualNetworks/virtualNetworkPeerings/read"
        ],
        "NotActions": [
      
        ],
        "AssignableScopes": [
          "/subscriptions/{customer-subscription-id}"
        ]
    }
    ```

3. Run the following Azure CLI commands to give Astronomer support temporary permissions to establish a VNet peering connection:

    ```sh
    # Add Astronomer Service Principal
    az ad sp create --id a67e6057-7138-4f78-bbaf-fd9db7b8aab0
    
    # Create a Custom role with permissions prepared in previous step
    az role definition create --role-definition ~/astro-vnet-peering-creator-role.json

    # Assign Custom role to the Astronomer Service Principal ({customer-subscription-id} has to be replaced with your value)
    az role assignment create \
    --assignee a67e6057-7138-4f78-bbaf-fd9db7b8aab0 \
    --role "Astro VNET Peering Contributor" \
    --scope "/subscriptions/{customer-subscription-id}"

    # Verify an assignment
    az role assignment list --assignee a67e6057-7138-4f78-bbaf-fd9db7b8aab0 --all -o table
    ```

> 4. Contact [Astronomer support](https://cloud.astronomer.io/open-support-request) to tell them that you have granted permissions to the Astronomer Service Principal. In addition, provide the following details in your request:

Reviewer 2: ```suggestion
4. Contact [Astronomer support](https://cloud.astronomer.io/open-support-request) to tell them that you granted permissions to the Astronomer Service Principal. In addition, provide the following details in your request:
```

    - Astro Cluster ID 
    - Azure Tenant ID and Subscription ID of the target VNet
    - Resource group ID
    - VNet ID for the peering

After receiving your request, Astronomer support creates a VNet peering connection between the two VNets. No other actions are required from you. Astronomer support will notify you when the connection is ready to use.

> When the network connection is confirmed, you can delete the temporary roles you created using command (`{customer-subscription-id}` has to be replaced with your value):

Reviewer 2: ```suggestion
When the network connection is confirmed, you can delete the temporary roles you created using the following command. Replace `{customer-subscription-id}` with your value:
```

    ```sh
    az role assignment delete --assignee a67e6057-7138-4f78-bbaf-fd9db7b8aab0 --role "Astro VNET Peering Contributor" --scope "/subscriptions/{customer-subscription-id}"
    ```

</TabItem>

<TabItem value="VHub peering">

:::info

> This connection option is only available for dedicated Astro Hosted clusters and Astro Hybrid.

Reviewer 2: ```suggestion
This connection option is not available for standard Astro Hosted clusters. It is available for dedicated clusters and Astro Hybrid.
```

> This connection option is not available for standard Astro Hosted clusters. It is available for dedicated clusters and Astro Hybrid.

Reviewer: The text here is different from the other info banners

> :::info
> 
> This connection option is not available for standard Astro Hosted clusters. It is available for dedicated clusters and Astro Hybrid.
> 
> :::

Reviewer 2: ```suggestion
:::info
This connection option is only available for dedicated Astro Hosted clusters and Astro Hybrid.
:::
```

> To set up a private connection between an Astro Virtual Network (VNet) and an Azure VHub, you can create a VHub peering connection. VHub peering ensures private and secure connectivity, reduces network transit costs, and attach Astro environment to centralized managed network.

Reviewer 2: ```suggestion
To set up a private connection between an Astro Virtual Network (VNet) and an Azure VHub, you can create a VHub peering connection. VHub peering ensures private and secure connectivity, reduces network transit costs, and attaches the Astro environment to a centralized managed network.
```

1. Retrieve the following information from the target Azure environment that you want to connect with:

    - Azure Tenant ID and Subscription ID.
    - VHub name.
    - Resource Group ID.
    - Optional. Firewall IP address if you use any on the VHub side.

> 2. Prepare JSON file `astro-vhub-peering-creator-role.json` with the following permisisons (`{customer-subscription-id}` has to be replaced with your value):

Reviewer 2: ```suggestion
2. Prepare a `astro-vhub-peering-creator-role.json` JSON file with the following permissions. Replace `{customer-subscription-id}` with your value:
```

    ```sh
    {
        "Name": "Astro VHub Peering Contributor",
        "IsCustom": true,
        "Description": "Can create VNET peering with Astro.",
        "Actions": [
            "Microsoft.Resources/subscriptions/resourceGroups/read",
            "Microsoft.Resources/subscriptions/read",
            "Microsoft.Network/virtualHubs/hubVirtualNetworkConnections/write",
            "Microsoft.Network/virtualHubs/read",
            "Microsoft.Network/virtualWans/virtualHubs/read"
        ],
        "NotActions": [
      
        ],
        "AssignableScopes": [
          "/subscriptions/{customer-subscription-id}"
        ]
    }
    ```

> 3. Run the following Azure CLI commands to give Astronomer support temporary permissions to establish a VHub peering connection:

Reviewer 2: ```suggestion
3. Run the following Azure CLI commands to give Astronomer Support temporary permissions to establish a VHub peering connection:
```
Do we want this in public docs?

    ```sh
    # Add Astronomer Service Principal
    az ad sp create --id a67e6057-7138-4f78-bbaf-fd9db7b8aab0
    
    # Create a Custom role with permissions prepared in previous step
    az role definition create --role-definition ~/astro-vhub-peering-creator-role.json

    # Assign Custom role to the Astronomer Service Principal ({customer-subscription-id} has to be replaced with your value)
    az role assignment create \
    --assignee a67e6057-7138-4f78-bbaf-fd9db7b8aab0 \
    --role "Astro VHub Peering Contributor" \
    --scope "/subscriptions/{customer-subscription-id}"

    # Verify an assignment
    az role assignment list --assignee a67e6057-7138-4f78-bbaf-fd9db7b8aab0 --all -o table
    ```

> 4. Contact [Astronomer support](https://cloud.astronomer.io/open-support-request) to tell them that you have granted permissions to the Astronomer Service Principal. In addition, provide the following details in your request:

Reviewer 2: ```suggestion
4. Contact [Astronomer support](https://cloud.astronomer.io/open-support-request) to tell them that you granted them permissions to the Astronomer Service Principal. In addition, provide the following details in your request:
```

    - Astro Cluster ID 
    - Azure Tenant ID and Subscription ID with a VHub
    - Resource group ID
    - VHub name and preferable name for the peering

>     - Optional. Firewall IP address if you use any on the VHub side.

Reviewer 2: ```suggestion
    - (Optional) Firewall IP address if you use any on the VHub side.
```

> After receiving your request, Astronomer support creates a VHub peering connection to Astro VNet. No other actions are required from you. Astronomer support will notify you when the connection is ready to use.

Reviewer: @afastronomer should this say from Astro VNet? Not to Astro VNet.

Author: When you do this type of the peering, every time you connect from Hub to Spokes. We login into customer environment and **from there** we initiate a connection.

Reviewer: Ah got it thanks.

> When the network connection is confirmed, you can delete the temporary roles you created using command (`{customer-subscription-id}` has to be replaced with your value):

Reviewer 2: ```suggestion
When the network connection is confirmed, you can delete the temporary roles you created using the following command. Replace `{customer-subscription-id}` with your value:
```

</TabItem>

<TabItem value="VPN">

:::info

> This connection option is only available for dedicated Astro Hosted clusters and Astro Hybrid.

Reviewer 2: ```suggestion
This connection option is not available for standard Astro Hosted clusters. It is available for dedicated clusters and Astro Hybrid.
```

> This connection option is not available for standard Astro Hosted clusters. It is available for dedicated clusters and Astro Hybrid.

Reviewer: The text here is different from the other info banners

> :::info
> 
> This connection option is not available for standard Astro Hosted clusters. It is available for dedicated clusters and Astro Hybrid.
> 
> :::

Reviewer 2: ```suggestion
:::info
This connection option is only available for dedicated Astro Hosted clusters and Astro Hybrid.
:::
```

> Use this connectivity type for access to on-premises resources or inside onther cloud providers.

Reviewer 2: ```suggestion
Use this connectivity type to access on-premises resources or resources in other cloud providers.
```

#### Perequisites

Retrieve the following information about your VPN device or application:

    - Public IP address

>     - Subnet CIDR (multipple if needed) on your side of the connection

Reviewer 2: ```suggestion
    - Subnet CIDR range, multiple if needed, for your side of the connection
```

    - Preferences regarding shared key and BGP usage
    - IKE settings for the tunnel

#### Contact Astronomer support for VPN configuration on Astro side

> Submit all collected details to [Astronomer support](https://cloud.astronomer.io/open-support-request). CRE team will proceed with the required steps. In case of additional questions or clarifications, CRE team will contact you within Support request or let you know about connectivity tests.

Reviewer 2: ```suggestion
Submit all collected details to [Astronomer support](https://cloud.astronomer.io/open-support-request). The Astronomer CRE team will proceed with the required steps. The CRE team will contact you using your support ticket to ask follow-up questions, request clarification, or let you know about connectivity tests.
```
